### DISCUSSION TITLE
Possible to use update from useSession on unauthenticated users?

### DISCUSSION THREAD

**Discussion by william-will-angi** [nextauthjs/next-auth#10899] [Help]
Is there a way to get the `update` function returned by `useSession` to run for unauthenticated users? I'm trying to use some code similar to this, but it doesn't fetch the `/api/auth/session` endpoint a second time the way I would expect.
```tsx
import {useEffect} from 'react';
import {useSession} from 'next-auth/react';

const useMyHook = () => {
  const session = useSession()
  useEffect(()=>{
    if (session.status === 'unauthenticated') {
      session.update();
    }
  }, [session.status])
}
```

I also tried the following, more involved code, but this also did not update the internal session stored by next-auth client-side.
```tsx
import {useEffect} from 'react';
import {useSession, getSession} from 'next-auth/react';

const manualSessionUpdate = async (updateFn) => {
  const newSession = await getSession()
  updateFn(newSession)
}

const useMyHook = () => {
  const session = useSession()
  useEffect(()=>{
    if (session.status === 'unauthenticated') {
      manualSessionUpdate(session.update);
    }
  }, [session.status])
}
```

The use case is being able to refresh the user's authentication status without requiring a page reload. For example, if you logged in with another window/tab.

**Comment by lmac-1** [comment]
> The use case is being able to refresh the user's authentication status without requiring a page reload. For example, if you logged in with another window/tab.

I am wanting to implement the same behaviour. @william-will-angi did you ever figure it out? I can't seem to find a way to get the user's authentication status without refreshing the page when you logged in another tab.

**Reply by william-will-angi** [reply]
@lmac-1 we ended up using the library `patch-package` to modify this line to exclude the check for empty sessions: https://github.com/nextauthjs/next-auth/blob/1fac7d3900463259617ba941d49da8cf70abc919/packages/next-auth/src/react/index.tsx#L472

I am not certain what side effects this might have although, fwiw, we have not noticed any major regressions since rolling that out
